### PR TITLE
Revisions to callback system

### PR DESCRIPTION
**Pull request by wesbonelli**
@MatthewGerber This is a partially tested first draft.

### CODE REVIEW

**Comment by MatthewGerber** [comment]
@wesbonelli I committed the alert exclusion stuff, which created conflicts with this. Think you can fix them?

**Comment by wesbonelli** [comment]
@MatthewGerber Sure, I'll take a look tomorrow.

File: Sensus.Tests.Cloud/Sensus.Tests.Cloud.csproj [hunk 1]
```diff
@@ -70,11 +70,5 @@
       <ReferenceOutputAssembly>False</ReferenceOutputAssembly>
       <Private>False</Private>
     </ProjectReference>
-    <ProjectReference Include="..\Sensus.Android\Sensus.Android.csproj">
-      <Project>{D33B6CDB-795C-4826-B131-47C526ACE07F}</Project>
-      <Name>Sensus.Android</Name>
-      <ReferenceOutputAssembly>False</ReferenceOutputAssembly>
-      <Private>False</Private>
-    </ProjectReference>
   </ItemGroup>
```
Comment (MatthewGerber) [inline Sensus.Tests.Cloud/Sensus.Tests.Cloud.csproj]: This shouldn't have been deleted.

Comment (wesbonelli) [inline Sensus.Tests.Cloud/Sensus.Tests.Cloud.csproj]: @MatthewGerber Sorry about that. I'm working on some tests for the callback system, but they're not commit-ready yet. I must have thought this was related to them.

On that note, I'm meeting with Mark to ask a few questions re the tests on Wednesday, and I'm hoping to have them added here by the end of the week.

**Comment by wesbonelli** [comment]
@MatthewGerber It turns out there are some changes to non-callback code that need to be made before we can run unit tests against this. I have some tests written, but I don't want to make any more significant changes before running them by you first, so I'm going to hold off for now.

**Comment by MatthewGerber** [comment]
Had to re-do this, given the conflicts. Completed by aacf737d3cc67c82d10d069ea739cf1701158674.

**Pull request closed without merging**